lib/web3/contract.js

```diff
@@ -207,33 +201,30 @@ ContractFactory.prototype.new = function () {
         options = args.pop();
     }
 
-    // throw an error if there are no options
-
     var bytes = encodeConstructorParams(this.abi, args);
     options.data += bytes;
 
-
-    if(callback) {
+    if (callback) {
 
         // wait for the contract address adn check if the code was deployed
-        web3.eth.sendTransaction(options, function (err, hash) {
+        this.web3.eth.sendTransaction(options, function (err, hash) {
             if (err) {
                 callback(err);
             } else {
                 // add the transaction hash
-                contract.transactionHash = hash;
+                contract._transactionHash = hash;
```

Comment:
This is a property people are supposed to access from the callback, we can't prefix that with `_`.

This is an unnecessary breaking change..